src/main/java/com/coronacarecard/trace/AddResponseHeaderFilter.java

```diff
@@ -0,0 +1,43 @@
+package com.coronacarecard.trace;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import brave.Tracer;
+import org.springframework.stereotype.Component;
+
+@WebFilter()
+@Component
+public class AddResponseHeaderFilter implements Filter {
+
+    @Autowired
+    Tracer tracer;
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException {
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+        httpServletResponse.setHeader("tracer-id", tracer.currentSpan().context().traceIdString());
+        chain.doFilter(request, response);
+    }
+
+    @Override
```

Comment:
these can be empty?

Reply:
We are not overriding init and destroy methods. Init is used for using FilterConfig for any special initiallization. And we are using default destroy no special need to override destroy as well. Below are some details about them.


void init(FilterConfig filterConfig)
The servlet container calls init() as a filter is first instantiated and placed into service. This method takes a javax.servlet.FilterConfig instance as input, which the servlet container uses to pass information to the filter during the initialization. Include any special initialization requirements in your implementation. Also see "Methods of the FilterConfig Interface".


destroy(): 
The servlet container calls destroy() after all execution of the filter has completed (all threads of the doFilter() method have completed, or a timeout has occurred) and the filter is being taken out of service. Include any special cleanup requirements in your implementation.